### PR TITLE
Date medium format

### PR DESCRIPTION
**Pull request by dgharde**
## Related Issue
Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
the issues fixed with this PR (eg Fixes #<github issue number>). If an issue doesn't yet exist please create one if reasonable to aid 
in issue tracking.

**NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
page above.

## Description
- InputDate now displays date in MediumFormat (eg: 19-Mar-2020)
System date format is fetched if the System Date is in any one of the following formats:
dd-MM-yyyy  or dd-MMM-yyyy
MM-dd-yyyy or  MMM-dd-yyyy
yyyy-MM-dd or yyyy-MMM-dd
yyyy-dd-MM or yyyy-dd-MMM

dd-> day (eg: 19)
MM-> month( eg: 01)
MMM-> month (eg: Jan)
yyyy-> year (eg: 2021)

- The selected date in calendar changes based on the input in the Text Field

- Action Submit returns only valid dates in yyyy-mm-dd format

## Sample Card
System Format: dd-MM-yyyy 
Inputted Date: Valid
image

System Format: MMM-dd-yyyy
Inputted Date: Invalid
image

System Format: yyyy-MM-dd
Inputted Date: Valid
image

System Format: yyyy-dd-MM
Inputted Date: Invalid
image

Calendar selected date being set based on Text Field's input

https://user-images.githubusercontent.com/78541663/111790228-fee25200-88e7-11eb-9113-343a27f9d275.mp4

 

## How Verified
How you verified the fix, including one or all of the following: 
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it 
will aid in code reviews or corresponding fixes on other platforms for eg.***

### CODE REVIEW

File: source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp [hunk 1]
```diff
@@ -1117,37 +1119,50 @@ namespace RendererQml
...
 				break;
 			}
 		}
 
+		uiDateInput->AddFunctions(Formatter() << "function setValidDate(dateString)" << "{"
+			<< "var Months = {Jan: 0,Feb: 1,Mar: 2,Apr: 3,May: 4,Jun: 5,July: 6,Aug: 7,Sep: 8,Oct: 9,Nov: 10,Dec: 11};"
+			<< "var d=new Date(" << year_Text << "," << "Months[" << month_Text << "]," << day_Text << ");"
+			<< "if( d.getFullYear().toString() === " << year_Text <<"&& d.getMonth()===Months[" << month_Text << "] && d.getDate().toString()===" << day_Text << ")"
+			<< "{" << "selectedDate = d.toLocaleString(Qt.locale(\"en_US\"),\"yyyy-MM-dd\");" << "}"
+			<< "else { selectedDate = '' };" << "}");
```
Comment (dgharde) [inline source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp:1163]: This function converts the text in the TextField to Date Object. 
To check if the Date Object is valid, we're checking if the day, month and year of the Date object is the same as the input used.

File: source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp [hunk 2]
```diff
@@ -1010,14 +1010,10 @@ namespace RendererQml
         uiDateInput->Property("font.pixelSize", std::to_string(fontSize));
         uiDateInput->Property("color", context->GetColor(AdaptiveCards::ForegroundColor::Default, false, false));
 
-        if (!input->GetValue().empty())
-        {
-            uiDateInput->Property("text", "\"" + Utils::GetDate(input->GetValue(), false) + "\"");
-        }
-
         uiDateInput->Property("property string selectedDate", "\"" + input->GetValue() + "\"");
+		uiDateInput->AddFunctions(Formatter() << "signal on_" << uiDateInput->GetId() << "TextChanged(var dateText)");
```
Comment (rj100189) [inline source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp:1014]: ideally, signal does not start as on*... 
```suggestion
		uiDateInput->AddFunctions(Formatter() << "signal " << uiDateInput->GetId() << "TextChanged(var dateText)");
```
```suggestion
		uiDateInput->AddFunctions(Formatter() << "signal on_" << uiDateInput->GetId() << "TextChanged(var dateText)");
```

**Review by rj100189 — APPROVED** [review]
minor feedback

File: source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp (hunk 2 above)
Comment (dgharde) [reply source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp]: Thanks for the feedback. I've made the change with the latest commit

**Review by rj100189 — APPROVED** [review]

**Pull request merged**